CMakeLists.txt

```diff
@@ -10,7 +10,27 @@ if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
     set(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT False)
 endif()
 
+# Set default build type to Release
+if(NOT CMAKE_BUILD_TYPE)
+  messageInfo("No CMAKE_BUILD_TYPE, using default CMAKE_BUILD_TYPE=Release")
+  set(CMAKE_BUILD_TYPE "Release")
+endif()
+
+# Disable in-source builds
+set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
+
+# Be nice to visual studio
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+# Be nice and export compile commands by default, this is handy for clang-tidy
+# and for other tools.
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+# Append library and executable names with d in debug mode
+set(CMAKE_DEBUG_POSTFIX d)
```

Comment:
I do not like to suffix debug artifacts. Instead of this, there are installed in a specific folder.
When you have scripts that build-test-analyse-verify-relocate ... the executable or the lib, I found easier to change the base directory of the files manipulated into the script than adding a suffix to each file (with some that do not have suffix and some who have ...)
```suggestion
```